app/org/sagebionetworks/bridge/services/EmailVerificationService.java

```diff
@@ -25,14 +28,42 @@
  */
 @Component
 public class EmailVerificationService {
+
     private static final Logger LOG = LoggerFactory.getLogger(EmailVerificationService.class);
     
+    private static final String KEY_POSTFIX = ":emailVerificationStatus";
+    
     private AmazonSimpleEmailServiceClient sesClient;
+    private CacheProvider cacheProvider;
 
     @Resource(name="sesClient")
     final void setAmazonSimpleEmailServiceClient(AmazonSimpleEmailServiceClient sesClient) {
         this.sesClient = sesClient;
     }
+    @Autowired
+    final void setCacheProvider(CacheProvider cacheProvider) {
+        this.cacheProvider = cacheProvider;
+    }
+    
+    private String getVerifiedAddressKey(String emailAddress) {
+        return emailAddress + KEY_POSTFIX;
+    }
+    
+    private EmailVerificationStatus cacheAndReturn(String emailAddress, EmailVerificationStatus status) {
+        String key = getVerifiedAddressKey(emailAddress);
+        cacheProvider.setString(key, status.name(), 60*60*24);
```

Comment:
Hmm, yes. This happens outside of our system. Okay, I've reduced it to 5 minutes. From testing I can tell you when you trigger it from the UI, the UI calls getEmailStatus() and this always updates the cache. So you notice no delay when setting up a study to work with it. However outside the UI you would have to wait.